telematic_system/telematic_cloud_messaging/src/main/java/com/telematic/telematic_cloud_messaging/message_converters/JSON2KeyValuePairsConverter.java

```diff
@@ -22,21 +22,29 @@ public String convertJson2KeyValuePairs(String json_str, List<String> to_str_val
             for (Object key : json.keySet()) {
                 key_count++;
                 Object value = json.get(key.toString());
+                boolean is_ignored = false;
 
                 if (value == null || value.toString().isEmpty())
                 {
                     pairs += key + "=\"NA\"";
                 }
                 else {
-                    boolean is_skip = false;
-                    for (String value_item: to_str_values)
+                    for(String field: ignore_fields)
                     {
-                        if (key.toString().strip().equals(value_item)){
+                        if(key.toString().strip().equalsIgnoreCase(field))   
+                        {
+                            is_ignored = true;
+                        }
+                    }
+                    boolean is_processed = false;
```

Comment:
Not a huge concern, but since we are processing all the fields, can we change this bool to something like is_str_cast?